File: backend/internal/model/invitation.go
package model

import (
	"time"
)

type OrganizationInvitation struct {
	ID             int       `gorm:"primaryKey;autoIncrement"`
	OrganizationID int       `gorm:"not null;index"`
	Email          string    `gorm:"size:255;not null;index"`
	Token          string    `gorm:"size:255;not null;uniqueIndex"`
	Role           string    `gorm:"size:20;default:'member'"`
	Status         string    `gorm:"size:20;default:'pending'"`
	ExpiresAt      time.Time `gorm:"not null"`
	CreatedBy      int       `gorm:"not null"`
	CreatedAt      time.Time `gorm:"autoCreateTime"`
	UpdatedAt      time.Time `gorm:"autoUpdateTime"`

	Organization Organization `gorm:"foreignKey:OrganizationID"`
	Creator      User         `gorm:"foreignKey:CreatedBy"`
}


const (
	InvitationStatusPending  = "pending"
	InvitationStatusAccepted = "accepted"
	InvitationStatusDeclined = "declined"
	InvitationStatusExpired  = "expired"
)

File: frontend/src/features/invitations/types/invitation.ts
export type InvitationStatus = 'pending' | 'accepted' | 'declined' | 'expired';
export type InvitationRole = 'admin' | 'member';

export interface Invitation {
    id: number;
    organization_id: number;
    email: string;
    token: string;
    role: InvitationRole;
    status: InvitationStatus;
    expires_at: string;
    created_by: number;
    created_at: string;
    updated_at: string;
    organization?: {
        id: number;
        name: string;
        slug: string;
    };
}

export interface CreateInvitationRequest {
    email: string;
    role: InvitationRole;
}

export interface InvitationInfo {
    organization_id: number;
    organization_name?: string;
    email: string;
    role: InvitationRole;
    status: InvitationStatus;
    expires_at: string;
}